### PR TITLE
TextField

### LINKED ISSUE #114: <TextField /> should support all Input's props
- <TextField /> should support all Input's props
- <Checkbox /> should have the `register` prop forwarded to its input.

### PR DESCRIPTION
**Pull request by vitalii-ilchenko-ssa-group**
Closes #114

### CODE REVIEW

File: packages/core/src/components/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -40,6 +40,7 @@ const Checkbox = ({
   }, [externalState]);
 
   const checkboxId = id || autoGenId;
+  const register = rest.register;
```
Comment (react-developer-ssa-group) [inline packages/core/src/components/Checkbox/Checkbox.tsx:43]: Do we need pass `register` to the CheckboxBase? If not, could we get register in the Checkbox props?
I mean the next approach:
`const Checkbox = ({ text, id, ...other props..., register, ...rest }: ICheckboxProps)`

File: packages/core/src/components/TextField/types.ts [hunk 2]
```diff
@@ -13,6 +13,7 @@ export interface TextFieldProps
   disabled?: boolean;
   rows?: number;
   endElement?: React.ReactElement;
+  startElement?: React.ReactElement;
```
Comment (arthur-kashaiev-ssa-group) [inline packages/core/src/components/TextField/types.ts:16]: I suggest extending this type from `InputProps`. What do you think?
TextField is a wrapper over Input. And currently, TextField has duplicate type definitions, which complicates support.

**Review by arthur-kashaiev-ssa-group — CHANGES_REQUESTED** [review]

File: packages/core/src/components/Checkbox/Checkbox.tsx (hunk 1 above)
Comment (vitalii-ilchenko-ssa-group) [reply packages/core/src/components/Checkbox/Checkbox.tsx]: Done

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 5  | ❗       |
    


*❗ storybook/components-icons--icons__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ storybook/components-icons--icons__[w1920px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ storybook/components-icons--icons__[w390px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ storybook/components-icons--icons__[w900px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ storybook/utility-responsive-image--responsive-image__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |

**Review by react-developer-ssa-group — APPROVED** [review]

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 1  | ❗       |
    


*❗ custom/widgets-user-profile--is-opened__[w1920px]*: 


| Before | After |
| --- | --- |
|  |  |

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 4  | ❗       |
    


*❗ storybook/components-icons--icons__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ storybook/components-icons--icons__[w1920px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ storybook/components-icons--icons__[w390px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ storybook/components-icons--icons__[w900px]*: 


| Before | After |
| --- | --- |
|  |  |

**Review by arthur-kashaiev-ssa-group — APPROVED** [review]

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 5  | ❗       |
    


*❗ custom/widgets-meal-nutrients--hint-opened__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ custom/widgets-meal-nutrients--hint-opened__[w390px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ custom/widgets-navbar-opened__[w1920px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ custom/widgets-progressInfo__[w1920px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ custom/widgets-user-profile--is-opened__[w1920px]*: 


| Before | After |
| --- | --- |
|  |  |

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 6  | ❗       |
    


<details>
<summary>❗ storybook/components-icons--icons__[w1440px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/components-icons--icons__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/components-icons--icons__[w390px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/components-icons--icons__[w900px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/utility-responsive-image--responsive-image__[w1440px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/utility-responsive-image--responsive-image__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 6  | ❗       |
    


<details>
<summary>❗ custom/widgets-collapsiblenavbar-opened__[w1439px]_statistics_submenu_popover (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ custom/widgets-collapsiblenavbar-opened__[w899px]_statistics_submenu (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ custom/widgets-meal-nutrients--hint-opened__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ custom/widgets-meal-nutrients--hint-opened__[w900px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ custom/widgets-navbar-opened__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ custom/widgets-progressInfo__[w900px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>

**Review by react-developer-ssa-group — APPROVED** [review]

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 1  | ❗       |
    


*❗ storybook/utility-responsive-image--responsive-image__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 7  | ❗       |
    


<details>
<summary>❗ custom/widgets-collapsiblenavbar-opened__[w1439px]_statistics_submenu_popover (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ custom/widgets-collapsiblenavbar-opened__[w899px]_statistics_submenu (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ custom/widgets-linkstabbar--current-focus__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ custom/widgets-meal-nutrients--hint-opened__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ custom/widgets-meal-nutrients--hint-opened__[w900px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ custom/widgets-user-profile--is-opened__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/widgets-heart-rate--default__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>

**Pull request merged**